private_gpt/components/ingest/ingest_component.py

```
@@ -102,6 +102,51 @@ def delete(self, doc_id: str) -> None:
             self._save_index()
 
 
+class CustomPdfFileIngestComponentWithIndex(BaseIngestComponentWithIndex):
+    """This is Emerson Collective's custom PDF file ingestion component."""
+
+    def __init__(
+        self,
+        storage_context: StorageContext,
+        service_context: ServiceContext,
+        *args: Any,
+        **kwargs: Any,
+    ) -> None:
+        super().__init__(storage_context, service_context, *args, **kwargs)
+
+    def ingest(self, file_name: str, file_data: Path) -> list[Document]:
+        logger.info("Custom Ingestion (PDF only) - Ingesting file_name=%s", file_name)
+        documents = IngestionHelper.transform_pdf_file_into_documents_and_save(
+            file_name, file_data
+        )
+        logger.info(
+            "Transformed file=%s into count=%s documents",
+            file_name,
+            len(documents),
+        )
+        logger.debug("Saving the documents in the index and doc store")
+        return self._save_docs(documents)
+
+    def bulk_ingest(self, files: list[tuple[str, Path]]) -> list[Document]:
+        logger.info(
+            "Bulk Ingestion (PDF only) - NOT implemented! Returning empty list.",
+        )
```

If this method is called anywhere, we should log a warning or an error here -- if not, raise an exception!

Otherwise, the method could potentially fail silently and make it more difficult to fix in the future.